src/overlays/actors/ovl_En_Fw/z_en_fw.c

```diff
@@ -21,39 +29,438 @@ const ActorInit En_Fw_InitVars = {
+        // not on the ground or moving upwards.
+        return false;
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+    this->bounceCnt--;
+    if (this->bounceCnt <= 0) {
+        if (this->bounceCnt == 0) {
+            this->bounceCnt = 0;
+            this->actor.velocity.y = 0.0f;
+            return true;
+        }
+        this->bounceCnt = totalBounces;
+    }
+    this->actor.velocity.y = yVelocity;
+    this->actor.velocity.y *= ((f32)this->bounceCnt / totalBounces);
+    return 1;
+}
+
+s32 EnFw_PlayerInRange(EnFw* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    CollisionPoly* poly;
+    u32 bgId;
+    Vec3f collisionPos;
+
+    if (this->actor.xzDistFromLink > 300.0f) {
+        return false;
+    }
+
+    if (ABS((s16)((f32)this->actor.yawTowardsLink - (f32)this->actor.shape.rot.y)) >= 0x1C71) {
+        return false;
+    }
+
+    if (func_8003DE84(&globalCtx->colCtx, &this->actor.posRot.pos, &player->actor.posRot.pos, &collisionPos, &poly, 1,
+                      0, 0, 1, &bgId)) {
+        return false;
+    }
+
+    return true;
+}
+
+Vec3f* EnFw_GetPosAdjAroundCircle(Vec3f* dst, EnFw* this, f32 radius, s16 dir) {
+    s16 angle;
+    Vec3f posAdj;
+
+    // increase rotation around circle ~30 degrees.
+    angle = Math_Vec3f_Yaw(&this->actor.parent->initPosRot.pos, &this->actor.posRot.pos) + (dir * 0x1554);
+    posAdj.x = (Math_Sins(angle) * radius) + this->actor.parent->initPosRot.pos.x;
+    posAdj.z = (Math_Coss(angle) * radius) + this->actor.parent->initPosRot.pos.z;
+    posAdj.x -= this->actor.posRot.pos.x;
+    posAdj.z -= this->actor.posRot.pos.z;
+    *dst = posAdj;
+    return dst;
+}
+
+s32 EnFw_CheckCollider(EnFw* this, GlobalContext* globalCtx) {
+    ColliderBody* body;
+    s32 phi_return;
+
+    if (this->collider.base.acFlags & 2) {
+        body = &this->collider.list[0];
+        if (body->acHitItem->toucher.flags & 0x80) {
+            this->lastDmgHook = true;
+        } else {
+            this->lastDmgHook = false;
+        }
+        this->collider.base.acFlags &= ~2;
+        if (Actor_ApplyDamage(&this->actor) <= 0) {
+            if (this->actor.parent->colChkInfo.health <= 8) {
+                func_80032C7C(globalCtx, &this->actor);
+                this->actor.parent->colChkInfo.health = 0;
+            } else {
+                this->actor.parent->colChkInfo.health -= 8;
+            }
+            this->returnToParentTimer = 0;
+        }
+        return true;
+    } else {
+        return false;
+    }
+}
+
+s32 EnFw_SpawnDust(EnFw* this, u8 timer, f32 scale, f32 scaleSpeed, s32 dustCnt, f32 radius, f32 xzAccel, f32 yAccel) {
+    Vec3f pos = { 0.0f, 0.0f, 0.0f };
+    Vec3f speed = { 0.0f, 0.0f, 0.0f };
+    Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    s16 angle;
+    s32 i;
+
+    pos = this->actor.posRot.pos;
+    pos.y = this->actor.groundY + 2.0f;
+    angle = ((Math_Rand_ZeroOne() - 0.5f) * 65536.0f);
+    i = dustCnt;
+    while (i >= 0) {
```

Comment:
can this be a for loop?
```suggestion
    for(i = dustCnt; i >=0; i--) {
```
Alternatively, it may be the ever popular `while(i --> 0)`

Reply:
no for